src/main/java/com/github/princesslana/eriscasper/util/Shard.java

```diff
@@ -0,0 +1,49 @@
+package com.github.princesslana.eriscasper.util;
+
+import com.github.princesslana.eriscasper.ErisCasperFatalException;
+import com.github.princesslana.eriscasper.data.immutable.Wrapper;
+import com.ufoscout.properlty.Properlty;
+import java.util.Optional;
+
+public class Shard implements Wrapper<Integer[]> {
+
+  private final int shardNumber;
+  private final int shardTotal;
+
+  public Shard(int shardNumber, int shardTotal) {
+    if (shardNumber >= shardTotal || shardTotal < 1 || shardNumber < 0) {
+      throw new ErisCasperFatalException(
+          "Could not create sharding with [" + shardNumber + ", " + shardTotal + "]");
+    }
+    this.shardNumber = shardNumber;
+    this.shardTotal = shardTotal;
+  }
+
+  @Override
+  public Integer[] unwrap() {
+    return new Integer[] {getShardNumber(), getShardTotal()};
+  }
+
+  public int getShardNumber() {
```

Comment:
Do we need an unwrap method along with the getters? Seems like one or the other should be redundant.

Reply:
Yeah I'll probably remove the getters.